File: java/src/main/java/it/gabrieletondi/telldontaskkata/useCase/OrderApprovalUseCase.java
package it.gabrieletondi.telldontaskkata.useCase;

import it.gabrieletondi.telldontaskkata.domain.Order;
import it.gabrieletondi.telldontaskkata.domain.OrderStatus;
import it.gabrieletondi.telldontaskkata.repository.OrderRepository;

public class OrderApprovalUseCase {
    private final OrderRepository orderRepository;

    public OrderApprovalUseCase(OrderRepository orderRepository) {
        this.orderRepository = orderRepository;
    }

    public void run(OrderApprovalRequest request) {
        final Order order = orderRepository.getById(request.getOrderId());

        if (order.isShipped()) {
            throw new ShippedOrdersCannotBeChangedException();
        }

        boolean approved = request.isApproved();
        if (approved && order.getStatus().equals(OrderStatus.REJECTED)) {
            throw new RejectedOrderCannotBeApprovedException();
        }

        if (!approved && order.getStatus().equals(OrderStatus.APPROVED)) {
            throw new ApprovedOrderCannotBeRejectedException();
        }

        if (approved) {
            order.approve();
        } else {
            order.reject();
        }
        orderRepository.save(order);
    }

}

File: php/src/UseCase/OrderApprovalUseCase.php
<?php

declare(strict_types=1);

namespace Pitchart\TellDontAskKata\UseCase;

use Pitchart\TellDontAskKata\Domain\Order;
use Pitchart\TellDontAskKata\Domain\OrderStatus;
use Pitchart\TellDontAskKata\Repository\OrderRepository;

class OrderApprovalUseCase
{
    private OrderRepository $repository;

    /**
     * @param OrderRepository $repository
     */
    public function __construct(OrderRepository $repository)
    {
        $this->repository = $repository;
    }

    /**
     * @throws RejectedOrderCannotBeApprovedException
     * @throws ApprovedOrderCannotBeRejectedException
     * @throws ShippedOrdersCannotBeChangedException
     */
    public function run(OrderApprovalRequest $request): void
    {
        $order = $this->repository->getById($request->getId());

        if ($order->isShipped()) {
            throw new ShippedOrdersCannotBeChangedException();
        }

        if ($request->isApproved()) {
            $order->approve();
        } else {
            $this->rejectOrder($order);
        }

        $this->repository->save($order);
    }

    private function rejectOrder(Order $order): void
    {
        if ($order->isApproved()) {
            throw new ApprovedOrderCannotBeRejectedException();
        }

        $order->setStatus(OrderStatus::Rejected);
    }
}
